crml/staking/src/rewards/mod.rs

```diff
@@ -335,6 +324,71 @@ impl<T: Trait> HandlePayee for Module<T> {
 	}
 }
 
+impl<T: Trait> Module<T>
+where
+	BalanceOf<T>: FixedPointOperand,
+{
+	/// Calculate all payouts of the current era as of right now. Then filter out those not relevant
+	/// validator-exposure sets by calling the "filter" function and execute the on_each_payout
+	/// function for each of non filtered payouts.
+	/// Return the total rewards calculated for the stakers at the time of this call paired with the
+	/// reward the development cut.
+	fn on_all_payouts<F, A>(
+		validator_commission_stake_map: &[(T::AccountId, Perbill, Exposure<T::AccountId, BalanceOf<T>>)],
+		filter: F,
+		mut on_each_payout: A,
+	) -> (BalanceOf<T>, BalanceOf<T>)
```

Comment:
```suggestion
	fn calculate_payouts_filtered<F, A>(
		validator_commission_stake_map: &[(T::AccountId, Perbill, Exposure<T::AccountId, BalanceOf<T>>)],
		filter: Option<F>,
	) -> (BalanceOf<T>, BalanceOf<T>, Iterator<Item=(AccountId, Balance))
```

This API could be improved by leaving iteration of payouts for the caller to handle.
Means caller doesn't need to know the details of this function and easier to follow logic at the call site:
```
let (_, _, payouts) = calculate_payouts_filtered(stake_map, None);
payouts.foreach(|account, amount| {
  // do whatever
});

let (_, _, payouts) = calculate_payouts_with_filtered(stake_map, Some(|v, e| ... ));
payouts.foreach(|account, amount| {
  // do whatever
});
```

Reply:
I have refactored this function according to your suggestion except for Option<F> which could make the implementation harder to read with not much performance improvement.